chain/network/src/store/schema/mod.rs

```diff
@@ -0,0 +1,279 @@
+/// Schema module defines a type-safe access to the DB.
+/// It is a concise definition of key and value types
+/// of the DB columns. For high level access see store.rs.
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_crypto::Signature;
+use near_network_primitives::time;
+use near_network_primitives::types as primitives;
+use near_primitives::account::id::{AccountId, ParseAccountError};
+use near_primitives::network::{AnnounceAccount, PeerId};
+use near_store::DBCol;
+use thiserror::Error;
+
+#[cfg(test)]
+mod tests;
+
+pub struct AccountIdFormat;
+impl Format for AccountIdFormat {
+    type T = AccountId;
+    fn to_vec(a: &AccountId) -> Vec<u8> {
+        a.as_ref().as_bytes().into()
+    }
+    fn from_slice(a: &[u8]) -> Result<AccountId, Error> {
+        std::str::from_utf8(a).map_err(Error::Utf8)?.parse().map_err(Error::AccountId)
+    }
+}
+
+#[derive(BorshSerialize, BorshDeserialize)]
+enum KnownPeerStatus {
+    Unknown,
+    NotConnected,
+    Connected,
+    /// UNIX timestamps in nanos.
+    Banned(primitives::ReasonForBan, u64),
+}
+
+impl From<primitives::KnownPeerStatus> for KnownPeerStatus {
+    fn from(s: primitives::KnownPeerStatus) -> Self {
+        match s {
+            primitives::KnownPeerStatus::Unknown => Self::Unknown,
+            primitives::KnownPeerStatus::NotConnected => Self::NotConnected,
+            primitives::KnownPeerStatus::Connected => Self::Connected,
+            primitives::KnownPeerStatus::Banned(r, t) => {
+                Self::Banned(r, t)
+            }
+        }
+    }
+}
+
+impl From<KnownPeerStatus> for primitives::KnownPeerStatus {
+    fn from(s: KnownPeerStatus) -> primitives::KnownPeerStatus {
+        match s {
+            KnownPeerStatus::Unknown => primitives::KnownPeerStatus::Unknown,
+            KnownPeerStatus::NotConnected => primitives::KnownPeerStatus::NotConnected,
+            KnownPeerStatus::Connected => primitives::KnownPeerStatus::Connected,
+            KnownPeerStatus::Banned(r, t) => primitives::KnownPeerStatus::Banned(
+                r,
+                t,
+            ),
+        }
+    }
+}
+
+#[derive(BorshSerialize, BorshDeserialize)]
+pub struct KnownPeerStateRepr {
```

Comment:
comment how this differs from KnownPeerState 
and why we're using this and not Borsch serialization directly.

Reply:
done